File: app/src/main/java/ru/geekbrains/githubclient/mvp/presenter/Presenter.java
package ru.geekbrains.githubclient.mvp.presenter;

import ru.geekbrains.githubclient.mvp.model.Model;
import ru.geekbrains.githubclient.mvp.view.MainView;

public class Presenter {
    private MainView view;
    private Model model = new Model();

    public Presenter(MainView view) {
        this.view = view;
    }


    public void counterClick(int id) {
        int newValue = model.getCurrent(id) + 1;
        model.set(id, newValue);
        view.setButtonText(id, String.valueOf(((newValue))));
    }
}

File: app/src/main/java/ru/geekbrains/githubclient/lesson4/Presenter.kt
package ru.geekbrains.githubclient.lesson4


import android.net.Uri

import io.reactivex.rxjava3.core.Observer
import io.reactivex.rxjava3.disposables.Disposable
import io.reactivex.rxjava3.observables.ConnectableObservable
import moxy.MvpPresenter

class Presenter(private val model: Model): MvpPresenter<View>() {

    fun convertImage(imagePath: Uri, imagesRoot: String) {

                val hotObservable: ConnectableObservable<Int> = model.convert(

                        imagePath,  // ссылка на картинку jpg
                        imagesRoot// путь к папке готового png

                )
                hotObservable.subscribe(object : Observer<Int> {
                    override fun onSubscribe(d: Disposable?) {
                        viewState.runConvert(imagePath)
                    }

                    override fun onNext(t: Int) {
                        //progressbar
                    }

                    override fun onError(e: Throwable?) {
                       e.toString()
                    }

                    override fun onComplete() {
                        model.convImage?.run {
                            viewState.finConvert(this)
                        }
                    }
                }
                )
                hotObservable.connect()
            }

        }
